### PR TITLE
support 'repo' scheme for add-ons (jsc#SLE-22578, jsc#SLE-24584)

### PR DESCRIPTION
**Pull request by wfeldt**
## Task

- https://trello.com/c/CAytlL3W
- https://jira.suse.com/browse/SLE-22578
- https://jira.suse.com/browse/SLE-24584

Support new `repo` URI scheme that works relative to the installation medium.

## See also

- https://github.com/yast/yast-packager/pull/624

### CODE REVIEW

File: src/lib/add-on/clients/add-on_auto.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 require "yast"
 require "installation/auto_client"
+require "uri"
+require "yast2/rel_url"
```
Comment (lslezak) [inline src/lib/add-on/clients/add-on_auto.rb:23]: It uses `AddOnProduct.GetAbsoluteURL` so this `require` should not be needed...

Comment (wfeldt) [reply src/lib/add-on/clients/add-on_auto.rb]: That's right.

**Review by lslezak — APPROVED** [review]
LGTM

**Comment by lslezak** [comment]
Ah, that `require "uri"` was actually correct...

**Comment by wfeldt** [comment]
It's not too late...

**Review by lslezak — APPROVED** [review]
:+1:

**Pull request merged**